core/src/main/java/io/ddf/content/ViewHandler.java

```diff
@@ -83,15 +83,13 @@ public DDF project(List<String> columnNames) throws DDFException {
     DDF projectedDDF = sql2ddf(String.format("SELECT %s FROM %%s", selectedColumns),
         String.format("Unable to project column(s) %s from table %%s", selectedColumns));
     //reserve factor information
-    List<Schema.Column> columns = this.getDDF().getSchema().getColumns();
```

Comment:
Just swap the way we check for Factor columns. This saves running time a bit because we loop over the projected DDF, not the original one which has more columns.